vm_eval.c

```diff
@@ -1615,6 +1615,23 @@ rb_each(VALUE obj)
 
 static VALUE eval_default_path;
 
+static VALUE
+get_eval_default_path(void)
+{
+    int location_lineno;
+    VALUE location_path = rb_source_location(&location_lineno);
+    if (!NIL_P(location_path)) {
+        return rb_fstring(rb_sprintf("(eval at %s:%d)", RSTRING_PTR(location_path), location_lineno));
+    }
+
+    static VALUE eval_default_path;
```

Comment:
Should not shadow the file scope variable; it is used in `rb_current_realfilepath` too.

```suggestion
```